index.js

```diff
@@ -137,9 +138,27 @@ module.exports = {
         'semi': ['error', 'always', {
             omitLastInOneLineBlock: true
         }],
-        // Allow for multiple method signatures in TS (which appear as dupe class members)
-        "no-dupe-class-members": "off",
-        "no-redeclare": "off",
-        "@typescript-eslint/no-redeclare": ['error', {ignoreDeclarationMerge: true}]
-    }
+    },
+    // Run typescript-specific rules only on typescript files
+    overrides: [
+        {
+            'parserOptions': {'project': ['./tsconfig.json']},
+            // Note: Remember to disable the base rules when enabling the ts version
+            rules: {
+                'no-unused-vars': 'off',
+                '@typescript-eslint/no-unused-vars': ['error', {
+                    ignoreRestSiblings: true,
+                    args: 'none'
+                }],
+                // Sets 'return-await' to 'in-try-catch' mode
+                'no-return-await': 'off',
+                '@typescript-eslint/return-await': ['error', 'in-try-catch'],
```

Comment:
Let's enable the base (JS) version also by default, so we have the checking in both languages.

I could be wrong, but it looks like the TS version of the rule goes a bit further in that it can _require_ an await when returning within a `try`.  The ESLint version will _allow_ you to `return await` or not when in a try, but doesn't seem to enforce it.  So some value still in enabling the TS version, but also value in enabling the base rule for JS files.